ffi/src/lib.rs

```diff
@@ -620,30 +620,136 @@ pub unsafe extern "C" fn version(snapshot: *const SnapshotHandle) -> u64 {
     snapshot.version()
 }
 
+/// The `EngineSchemaVisitor` defines a visitor system to allow engines to build their own
+/// representation of a schema from a particular schema within kernel.
+///
+/// The model is list based. When the kernel needs a list, it will ask engine to allocate one of a
+/// particular size. Once allocated the engine returns an `id`, which can be any identifier the
```

Comment:
nit: reading "which can be any identifier the engine wants" I thought this would be an opaque type, but IIUC we are assuming `usize`?

Reply:
yeah, I was thinking that "identifier" was enough to make it clear, but i've amended to ``"integer identifier ([`usize`])"`` which is hopefully more clear (and links to the usize docs so implementers can understand what type it will be in c/c++)